lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -27,7 +27,7 @@ def call(env)
 
             processor = Datadog::AppSec.processor
 
-            return @app.call(env) unless processor.ready?
+            return @app.call(env) unless processor
```

Comment:
To improve thread safety, Processor might be finalized already but not yet nil:

```suggestion
            return @app.call(env) if processor.nil? || !processor.ready?
```